cpp/src/slice2cpp/Gen.cpp

```diff
@@ -2743,12 +2758,20 @@ Slice::Gen::InterfaceVisitor::visitOperation(const OperationPtr& p)
 
     DocCommentPtr comment = p->parseDocComment(cppLinkFormatter);
 
+    string isConst = p->hasMetadata("cpp:const") ? " const" : "";
+    string noDiscard = "";
+
     if (ret)
     {
         string typeS = inputTypeToString(ret, p->returnIsOptional(), interfaceScope, p->getMetadata(), _useWstring);
         responseParams.push_back(typeS + " " + returnValueParam);
         responseParamsDecl.push_back(typeS + " ret");
         responseParamsImplDecl.push_back(typeS + " ret");
+
+        if (!amd && !isConst.empty())
+        {
+            noDiscard = "[[nodiscard]] ";
```

Comment:
What is the relation of nodiscard and const?

Reply:
Added comment.